### PR TITLE
PW-1887: fix for pms(afterpaytouch) with checkoutIssuerDirectLinkEnabled

### PR DESCRIPTION
**Pull request by Aleffio**

### CODE REVIEW

File: Gateway/Validator/CheckoutResponseValidator.php [hunk 1]
```diff
@@ -151,6 +151,10 @@ public function validate(array $validationSubject)
                         $paymentData = $response['paymentData'];
                     }
 
+                    if (!empty($response['details'])) {
+                        $details = $response['details'];
+                    }
```
Comment (rikterbeek) [inline Gateway/Validator/CheckoutResponseValidator.php]: why doing this here ? I only see this being used in the else can we combine it with that if ?

File: Controller/Process/Result.php [hunk 2]
```diff
@@ -389,7 +389,21 @@ protected function validatePayloadAndReturnResponse($response)
...
+            $key = array_search('returnUrlQueryString', $details[0]);
+
+            if ($key !== false) {
+                $request['details']['returnUrlQueryString'] = http_build_query($response);
+            } else {
+                $request["details"] = $response;
+            }
+        } else {
+            $request["details"] = $response;
+        }
```
Comment (rikterbeek) [inline Controller/Process/Result.php]: a bit concerned about this flow, is this something our merchants need ?

Comment (cyattilakiss) [reply Controller/Process/Result.php]: Yes unfortunately for checkout directIssuerPages we need to implement this extra step - so currently this is a quick fix for those who use this setting but in the future this account setting can become default and can be that the API will also adjust.
For now I would just use it like this and let's follow up with the checkout team if they are going to make it more generic later.

File: Gateway/Validator/CheckoutResponseValidator.php [hunk 3]
```diff
@@ -184,6 +184,9 @@ public function validate(array $validationSubject)
                         // Flag to show we are in the checkoutAPM flow
                         $payment->setAdditionalInformation('checkoutAPM', true);
 
+                        if (!empty($details)) {
```
Comment (rikterbeek) [inline Gateway/Validator/CheckoutResponseValidator.php]: where is this variable declared ?

File: Controller/Process/Result.php (hunk 2 above)
Comment (rikterbeek) [reply Controller/Process/Result.php]: We can declare $request["details"] = $response; first and then override it with the if statement. That results in 2 else statements we can remove and cleaner code. What do you think ?

**Pull request merged**